lib/install/proto/proto.go

```diff
@@ -69,7 +69,12 @@ func KeyToProto(key ops.SiteOperationKey) *OperationKey {
 
 // HasSpecificPhase determines if this request is for a specific phase (other than root)
 func (r *ExecuteRequest) HasSpecificPhase() bool {
-	return r.Phase != nil && !r.Phase.IsResume()
+	return !r.HasResume()
+}
+
+// HasResume determines if this is a request to resume an operation
+func (r *ExecuteRequest) HasResume() bool {
+	return r.Phase == nil || r.Phase.IsResume()
```

Comment:
That's a little weird that "has resume" returns true if the phase is empty. Shouldn't this be `r.Phase != nil && r.Phase.IsResume()`?

Reply:
Resuming is starting the operation from scratch and skipping the phases already marked as completed. When starting from scratch - the phase is empty.